GVFS/GVFS.Common/Database/PlaceholderTable.cs

```diff
@@ -166,7 +169,16 @@ public void Remove(string path)
                 {
                     command.CommandText = "DELETE FROM Placeholder WHERE path = @path;";
                     command.AddParameter("@path", DbType.String, path);
-                    command.ExecuteNonQuery();
+
+                    try
+                    {
+                        this.writeToDatabaseResetEvent.WaitOne();

```

Comment:
What was the reason for using `AutoResetEvent` instead of `lock`?

I look around a bit and saw a discussion [here](https://stackoverflow.com/questions/17273933/autoresetevent-as-a-lock-replacement-in-c) that `AutoResetEvent` does not provide the same fairness guarantees as `lock`:

> WaitHandles do not provide such a guarantee. Which is very detrimental if the lock is contended, the same thread can acquire it repeatedly and other threads can starve forever.
>
> Use an appropriate synchronization object for locks. Wait handles should only be used for signaling.

Is that something we need to be concerned about?

Reply:
I had thought that locks were more heavy weight but I ran the performance tests using a `lock` and there doesn't seem to be a difference in the performance at least in what our performance tests are doing.  So I changed to use a lock since that seems to be simpler.